src/core/thread/mle.cpp

```diff
@@ -3032,6 +3032,9 @@ otError Mle::HandleLeaderData(const Message &aMessage, const Ip6::MessageInfo &a
         // if received timestamp does not match the local value and message does not contain the dataset,
         // send MLE Data Request
         if ((timestamp == NULL || timestamp->Compare(activeTimestamp) != 0) &&
+#if OPENTHREAD_FTD
```

Comment:
nit: this seems not readable. How about defining `IsLeader` like below:
```C++
#if OPENTHREAD_FTD
    ...
#else
    bool IsLeader() const { return false; }
#endif 
```

I think the compiler can optimize out `IsLeader` when `OPENTHREAD_FTD` is not defined, so the code side should not affected.